asllib/doc/ASLTypingReference.tex

```diff
@@ -2343,401 +2787,431 @@ \subsection{Comments}
         \item $\vt$ is an unconstrained integer;
-        \end{itemize}
+      \end{itemize}
       \item All of the following apply:
-        \begin{itemize}
-        \item \texttt{t1} has the structure of an integer;
-        \item \texttt{t2} has the structure of an unconstrained integer;
+      \begin{itemize}
+        \item $\vtone$ has the structure of an integer;
+        \item $\vttwo$ has the structure of an unconstrained integer;
         \item $\vt$ is an unconstrained integer;
-        \end{itemize}
-      \item One of the following applies:
-       \begin{itemize}
+      \end{itemize}
       \item One of the following applies:
-       \begin{itemize}
-       \item All of the following apply:
+      \begin{itemize}
+        \item One of the following applies:
+        \begin{itemize}
+          \item All of the following apply:
           \begin{itemize}
-          \item \texttt{t1} has the structure of an under-constrained integer;
-          \item \texttt{t2} has the structure of a constrained integer;
-          \item $\vt$ is an under-constrained integer;
+            \item $\vtone$ has the structure of an under-constrained integer;
+            \item $\vttwo$ has the structure of a constrained integer;
+            \item $\vt$ is an under-constrained integer;
           \end{itemize}
-        \item All of the following apply:
+          \item All of the following apply:
           \begin{itemize}
-          \item \texttt{t1} has the structure of a constrained integer;
-          \item \texttt{t2} has the structure of an under-constrained integer;
-          \item $\vt$ is an under-constrained integer;
+            \item $\vtone$ has the structure of a constrained integer;
+            \item $\vttwo$ has the structure of an under-constrained integer;
+            \item $\vt$ is an under-constrained integer;
           \end{itemize}
-       \end{itemize}
-         \item All of the following apply:
-           \begin{itemize}
-           \item \texttt{t1} has the structure of a well-constrained integer;
-           \item \texttt{t2} has the structure of a well-constrained integer;
-	       \item $\vt$ is a constrained integer whose constraint is calculated by
-	       applying the operation to all possible value pairs;
-           \end{itemize}
-      \item All of the following apply:
+        \end{itemize}
+        \item All of the following apply:
         \begin{itemize}
-        \item \texttt{t1} has the structure of \texttt{real};
-        \item \texttt{t2} has the structure of \texttt{real};
-        \item \texttt{op} is \texttt{PLUS}, \texttt{MINUS} or \texttt{MUL};
-        \item $\vt$ is \texttt{real};
+          \item $\vtone$ has the structure of a well-constrained integer;
+          \item $\vttwo$ has the structure of a well-constrained integer;
+	        \item $\vt$ is a constrained integer whose constraint is calculated by
+	        applying the operation to all possible value pairs;
```

Comment:
have you given the operation a name? if yes maybe it would be useful to use its name in this sentence, something like "applying the operation op to all possible pairs of values"?